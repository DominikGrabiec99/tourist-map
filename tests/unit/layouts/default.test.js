import { describe, it } from 'vitest';
import { mount } from '@vue/test-utils';

import { createPinia, setActivePinia } from 'pinia';

import defaultLayout from '@/layouts/default.vue';

import EStorage from '@/ts/enums/Storage';

const getItemSpy = vi.spyOn(Storage.prototype, 'getItem');
const setItemSpy = vi.spyOn(Storage.prototype, 'setItem');

describe('defaultLayout', () => {
  let wrapper;

  beforeEach(() => {
    setActivePinia(createPinia());
    wrapper = mount(defaultLayout);
  });

  afterEach(() => {
    sessionStorage.clear();
    getItemSpy.mockClear();
    setItemSpy.mockClear();
  });

  it('render layout', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should set currencies inStorage on mounted component', () => {
    expect(getItemSpy).toHaveBeenCalledWith(EStorage.CURRENCIES);
    expect(setItemSpy).toHaveBeenCalledWith(EStorage.CURRENCIES, 'true');
  });

  it('should toggle showCurrencyView', async () => {
    expect(wrapper.vm.showCurrencyView).toBeTruthy();
    expect(setItemSpy).toHaveBeenCalledWith(EStorage.CURRENCIES, 'true');
    await wrapper.vm.toggleCurrencyView(false);
    expect(wrapper.vm.showCurrencyView).toBeFalsy();
    expect(setItemSpy).toHaveBeenCalledWith(EStorage.CURRENCIES, 'false');
  });
});
