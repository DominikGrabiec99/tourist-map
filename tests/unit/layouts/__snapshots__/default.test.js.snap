// Vitest Snapshot v1

exports[`defaultLayout > render layout 1`] = `
<div
  class="min-h-screen bg-white dark:bg-neutral-900 transition-color"
>
  <!--v-if-->
  <organismdefaultmobileheader
    class="block md:hidden"
  />
  <organismdefaultheader
    class="hidden md:block"
  />
  <div
    class="container mx-auto px-2"
  >
    
    
  </div>
</div>
`;

exports[`defaultLayout > should set currencies inStorage 1`] = `
<div
  class="min-h-screen bg-white dark:bg-neutral-900 transition-color"
>
  <!--v-if-->
  <organismdefaultmobileheader
    class="block md:hidden"
  />
  <organismdefaultheader
    class="hidden md:block"
  />
  <div
    class="container mx-auto px-2"
  >
    
    
  </div>
</div>
`;
